packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,1109 @@
+                await waitForUpdatesAsync();
+                focusedElement = currentFocusedElement();
+                expect(focusedElement).toBeInstanceOf(TableRow);
+                const newRowIndex = (focusedElement as TableRow).rowStateIndex!;
+
+                expect(newRowIndex).toBeCloseTo(
+                    oldRowIndex - element.virtualizer.pageSize,
+                    0
+                );
+            });
+
+            it('with a row focused, if the table is scrolled away from that row, then back to it, it is focused again', async () => {
+                await sendKeyPressesToTable(
+                    keyArrowLeft,
+                    keyArrowDown,
+                    keyArrowDown,
+                    keyArrowDown
+                );
+                let focusedElement = currentFocusedElement();
+                expect(focusedElement).toBeInstanceOf(TableRow);
+                const originalFocusedRowId = (focusedElement as TableRow)
+                    .recordId;
+
+                await pageObject.scrollToLastRowAsync();
+                await pageObject.scrollToFirstRowAsync();
+
+                focusedElement = currentFocusedElement();
+                expect(focusedElement).toBeInstanceOf(TableRow);
+                expect((focusedElement as TableRow).recordId).toBe(
+                    originalFocusedRowId
+                );
+            });
+
+            it('with a cell focused, if the table is scrolled away from that cell, then back to it, it is focused again', async () => {
+                await sendKeyPressesToTable(
+                    keyArrowDown,
+                    keyArrowDown,
+                    keyArrowDown
+                );
+                let focusedElement = currentFocusedElement();
+                expect(focusedElement).toBeInstanceOf(TableCell);
+                const originalFocusedCellId = (focusedElement as TableCell)
+                    .recordId;
+
+                await pageObject.scrollToLastRowAsync();
+                await pageObject.scrollToFirstRowAsync();
+
+                focusedElement = currentFocusedElement();
+                expect(focusedElement).toBeInstanceOf(TableCell);
+                expect((focusedElement as TableCell).recordId).toBe(
+                    originalFocusedCellId
+                );
+            });
+        });
+
+        it('for a simple table with no columns sortable, on initial table focus, the 1st row is focused', async () => {
+            await setupBasicTable();
+            column1.sortingDisabled = true;
+            column2.sortingDisabled = true;
+            column3.sortingDisabled = true;
+            await waitForUpdatesAsync();
+            element.focus();
+            await waitForUpdatesAsync();
+
+            const firstRow = pageObject.getRow(0);
+            expect(currentFocusedElement()).toBe(firstRow);
+        });
+
+        describe('for a single-selection table with focus', () => {
```

Comment:
Maybe a degenerate case, but could we add a test that focusing a row and pressing space on a table with selection mode of "none" doesn't select that row?

Reply:
Added that test (and fixed a bug with that case, actually)